src/graph/VectorTilesService.ts

```diff
@@ -0,0 +1,89 @@
+/// <reference path="../../typings/browser.d.ts" />
+
+import * as rx from "rx";
+
+import {APIv3} from "../API";
+import {IGoogleTile, GoogleTiles} from "../Geo";
+import {MapillaryObject, MapillaryRect, Node} from "../Graph";
+
+export class VectorTilesService {
+    private _apiV3: APIv3;
+
+    private _googleTiles: GoogleTiles;
+    private _cacheNode$: rx.Subject<Node> = new rx.Subject<Node>();
+    private _mapillaryObjects$: rx.Observable<any>;
+
+    constructor (apiV3: APIv3) {
+        this._apiV3 = apiV3;
+        this._googleTiles = new GoogleTiles();
+
+        this._mapillaryObjects$ = this._cacheNode$.map<IGoogleTile>((node: Node): IGoogleTile => {
+            return this._googleTiles.getTileAtLatLon(node.latLon, 19);
+        }).distinct((tile: IGoogleTile): string => {
+            return tile.z + "-" + tile.x + "-" + tile.y;
+        }).flatMap((tile: IGoogleTile): any => {
+            let path: string = "tile['all']";
+
+            path += "[" + tile.z + "]";
+            path += "[" + tile.x + "]";
+            path += "[" + tile.y + "]";
+            path += "['objects'][0..100]['key', 'value', 'package', 'l', 'alt', 'rects', 'last_seen_at', 'first_seen_at']";
+
+            return this._apiV3.model.get(path);
+        }).flatMap<MapillaryObject>((tile: any): rx.Observable<MapillaryObject> => {
+            let z: number = parseInt(Object.keys(tile.json.tile.all)[0], 10);
+            let x: number = parseInt(Object.keys(tile.json.tile.all[z])[0], 10);
+            let y: number = parseInt(Object.keys(tile.json.tile.all[z][x])[0], 10);
+            let objects: any = tile.json.tile.all[z][x][y].objects;
+
+            let ret: MapillaryObject[] = [];
+
+            for (let object in objects) {
+                if (objects.hasOwnProperty(object)) {
+                    if (object !== "$__path") {
```

Comment:
The $__path property is handled differently here and in the DetectionComponent.
Also indentation can be decreased by using `!objects.hasOwnProperty(object)` and `continue`.